### PR TITLE
separate errors and fix a bug in grade

### PR DESCRIPTION
**Pull request by liadaram1**
In addition to separate the errors on course number, I fixed a bug when a user tries to remove the grade (for changing the course state to inProgress) it won't trigger an error.

### CODE REVIEW

File: packages/sogrim-app/src/components/Pages/SemestersPage/SemesterGrid/course-validator.ts [hunk 1]
```diff
@@ -1,22 +1,18 @@
 import { courseGradeOptions, emptyRow, RowData } from "../SemesterTabsConsts";
 
-export const validCourseNumber = (
+export const validCourseNumber = (courseNumber: string) => {
+  return /^\d+$/.test(courseNumber) && courseNumber.length === 6;
+};
+
+export const UniqueCourseNumber = (
```
Comment (benny-n) [inline packages/sogrim-app/src/components/Pages/SemestersPage/SemesterGrid/course-validator.ts:7]: camelCase

File: packages/sogrim-app/src/components/Pages/SemestersPage/SemesterGrid/course-validator.ts [hunk 2]
```diff
@@ -29,7 +25,13 @@ const validCourseCredit = (credit: string | number) => {
 
 const validGrade = (grade: any) => {
   const gradeNumber = Number(grade);
-  if (grade === "" || grade === 0 || grade === "0" || grade === "-") {
+  if (
+    grade === undefined ||
```
Comment (benny-n) [inline packages/sogrim-app/src/components/Pages/SemestersPage/SemesterGrid/course-validator.ts:29]: did you validate this? last time I added the check for "undefined" it caused some weird bugs when I tried adding\editing courses

**Review by benny-n — CHANGES_REQUESTED** [review]

File: packages/sogrim-app/src/components/Pages/SemestersPage/SemesterGrid/course-validator.ts (hunk 2 above)
Comment (liadaram1) [reply packages/sogrim-app/src/components/Pages/SemestersPage/SemesterGrid/course-validator.ts:29]: Yes I did. And it seems fine.
Do you have a concrete example so I can test it?

Comment (benny-n) [reply packages/sogrim-app/src/components/Pages/SemestersPage/SemesterGrid/course-validator.ts:29]: nothing specific, if you validated this it's ok from my end 👍

**Pull request merged**